### PR TITLE
Add a sitemap to pubs warehouse

### PR DESCRIPTION
**Pull request by jkreft-usgs**
our sitemap.xml was manually created long ago, and is now actually gone...so this is to recreate it dynamically.

### CODE REVIEW

File: pubs_ui/pubswh/views.py [hunk 1]
```diff
@@ -659,3 +660,52 @@ def unapi():
...
+def sitemap_list(year):
+    """
+    This function grabs content from the streaming service and generates a list that can be used to make
+    sitemap.xml data
+    :param year: The year that were are interested in
+    :return: a rendered sitemap.xml file
+    """
+    # there are only a few hundred pubs
+    if year == 1900:
+        pubs = get(pub_url + "publication/", params={"mimeType": "json", "endYear": year, "page_size": 5000},
```
Comment (mbucknell) [inline pubs_ui/pubswh/views.py]: Consider making this a little more DRY by creating a "params" dictionary with the mimeType and page_size and then adding the appropriate query parameter. And then make a single get call. 

Another question, why are you using page_size? Don't you want all of the pubs for a year?

File: pubs_ui/pubswh/views.py [hunk 2]
```diff
@@ -659,3 +660,52 @@ def unapi():
...
+                   verify=verify_cert, stream=True)
+    else:
+        pubs = get(pub_url + "publication/", params={"mimeType": "json", "year": year, "page_size": 5000},
+                   verify=verify_cert, stream=True)
+    records_list = []
+    for line in pubs.iter_lines():
```
Comment (mbucknell) [inline pubs_ui/pubswh/views.py]: Does pubs.iter_lines() work if a non 200 response occurs? Consider handling the error statuses explicitly.

File: pubs_ui/pubswh/views.py [hunk 3]
```diff
@@ -659,3 +660,52 @@ def unapi():
...
+        # filter out keep-alive new lines
+        if line:
```
Comment (mbucknell) [inline pubs_ui/pubswh/views.py]: This is confusing. Does requests iter_lines() filter the keep alives for you and are we actually sending keep alives? I think you could do this a little differently by doing pubs.iter_lines in a while loop until there is a good line, that will be the first line. And then use the for loop to grab the record from the line. I'm not a big fan of using exceptions to catch conditions that happen in the normal course of processing a record.

Is streaming really necessary here? We don't stream the response and if we grabbed the whole response at once the code would be simplier.

File: pubs_ui/pubswh/views.py (hunk 2 above)
Comment (jkreft-usgs) [inline pubs_ui/pubswh/views.py]: if there isn't a 200, it throws an ugly 500, which is good enough for these machine-focused pages

File: pubs_ui/pubswh/views.py (hunk 3 above)
Comment (jkreft-usgs) [inline pubs_ui/pubswh/views.py]: streaming cuts down the original page load size, but it is really pointless to optimize if we don't need to

File: pubs_ui/pubswh/views.py (hunk 1 above)
Comment (jkreft-usgs) [inline pubs_ui/pubswh/views.py]: there isn't a way to download everything in json, and 5000 covers the largest year by a good margin

**Pull request merged**